lib/kms.js

```
@@ -6,10 +6,32 @@ import * as brHttpsAgent from 'bedrock-https-agent';
 import {getAppIdentity} from 'bedrock-app-identity';
 import {KeystoreAgent, KmsClient} from '@digitalbazaar/webkms-client';
 
+const {util: {BedrockError}} = bedrock;
+
+/**
+ * Creates a new WebKMS keystore for tokenizer use.
+ * Requires bedrock.config.tokenizer.kms.baseUrl to be set.
+ *
+ * @param {object} options - Options hashmap.
+ * @param {string} options.controller - Keystore controller DID.
+ * @param {string} [options.kmsModule] - KMS module type.
+ * @param {string} [options.meterId] - Meter ID.
+ * @param {string} [options.invocationSigner]
+ * @param {string} [options.referenceId]
+ * @param {boolean} [options.applyIpAllowList=true]
+ *
+ * @return {Promise<object>} Resolves to the configuration for the newly
+ *   created keystore.
+ */
 export async function createKeystore({
   controller, kmsModule, meterId, invocationSigner, referenceId,
   applyIpAllowList = true
 } = {}) {
+  if(!(bedrock.config && bedrock.config.tokenizer &&
+    bedrock.config.tokenizer.kms && bedrock.config.tokenizer.kms.baseUrl)) {
+    throw new TypeError(
+      'bedrock.config.tokenizer.kms.baseUrl config value is required.');
+  }
```

I think this sort of thing should be handled on startup, paying the penalty one time, there is no need to run this check on every `createKeystore` operation.

A handler here for `bedrock.init` would ensure that the handlers in `bedrock-config-yaml` have done their work in a deployment situation.  https://github.com/digitalbazaar/bedrock-config-yaml/blob/main/lib/main.js#L16-L34

https://github.com/digitalbazaar/bedrock-ledger-agent/blob/master/lib/api.js#L138

The handler should simply throw if the config is not set properly which will prevent startup.

---

Thanks, moved it to a `bedrock.init` event, at the top.